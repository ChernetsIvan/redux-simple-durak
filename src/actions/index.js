import * as actionTypes from "./../constants/ActionTypes";

export const clickOnStartGameButton = () => ({
  type: actionTypes.START_GAME_CLICKED
});
export const clickOnBeginGameButton = () => ({
  type: actionTypes.BEGIN_GAME_CLICKED
});
export const clickOnRadioWhoMoveFirst = isFirstMovePlayer => ({
  type: actionTypes.WHO_MOVE_FIRST_CHANGED,
  payload: isFirstMovePlayer
});

export const clickOnCard = cardId => ({
  type: actionTypes.SOME_PLAYERS_CARD_CLICKED,
  payload: cardId
});

export const clickOnPlayerTake = () => ({
  type: actionTypes.PLAYER_TAKE_CLICKED
});
export const clickOnAiTake = () => ({
  type: actionTypes.AI_TAKE_CLICKED
});
export const clickOnRemoveCards = () => ({
  type: actionTypes.REMOVE_CARDS_CLICKED
});

export const gameOver = gameMode => ({
  type: actionTypes.GAME_OVER,
  payload: gameMode
});
