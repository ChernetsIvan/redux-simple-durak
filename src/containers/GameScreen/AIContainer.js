import { connect } from "react-redux";
import AI from "./../../components/GameScreen/AI";

const mapStateToProps = store => ({
  cards: store.game.computerCards
});

export default connect(mapStateToProps)(AI);
