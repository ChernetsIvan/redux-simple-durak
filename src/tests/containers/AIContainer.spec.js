import React from "react";
import renderer from "react-test-renderer";

import { Provider } from "react-redux";
import { createStore } from "redux";

import AIContainer from "./../../containers/GameScreen/AIContainer";
import reducer from "./../../reducers";
import { clickOnStartGameButton } from "./../../actions";

describe("AIContainer snapshot testing", () => {
  const store = createStore(reducer);

  test("Store does not contain any card", () => {
    const component = renderer.create(
      <Provider store={store}>
        <AIContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  store.dispatch(clickOnStartGameButton);

  test("Store contains some cards", () => {
    const component = renderer.create(
      <Provider store={store}>
        <AIContainer />
      </Provider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
});
