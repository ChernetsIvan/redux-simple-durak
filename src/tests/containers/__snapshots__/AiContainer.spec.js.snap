// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AIContainer snapshot testing Store does not contain any card 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col-10"
    >
      <div
        className="container invisible"
      >
        <div
          className="row"
        >
          <div
            className="col-1 card mb-1 shirt"
          />
        </div>
      </div>
    </div>
    <div
      className="col-1"
    />
  </div>
</div>
`;
